ramips: clear depends on mt76xx drivers

ptpt52 opened the pull request:
- drop depends on kmod-mt76 by default
 - each device individually selects the required mt76xx
 - some devices do not need drivers for the mt76xx series at all
 - since mt7610e is now supported, early devices added mt76x0e dependencies

 we can find the hardware specifies and determine the dependencies on these sites:
  https://wikidevi.com/wiki/
  https://openwrt.org/toh/hwdata/

 The following devices use mt76x0e:
  dir-810l
  tiny-ac
  mzk-750dhp
  tplink_c2-v1
  tplink_c20-v1
  ArcherC20i
  ArcherMR200
  tplink_c20-v4
  tplink_tl-wr902ac-v3
  rt-ac51u
  gl-mt750
  hc5761
  hc5861 (not sure)

ptpt52 commented:
I updated the changes and made precise adjustments to more devices depending on mt76.

mkresin commented:
I'm fine with the generic approach. Would you please rebase the PR to resolve the merge conflicts and take care of boards added in the meantime.

Would you please elaborate how you gathered the information which board requires a particular driver or no driver. It would make it way easier to ensure that all boards are covered and we don't introduce any kind of regression.

ptpt52 commented:
I update the info and rebase to the master HEAD

ptpt52 commented:
I can't be 100% correct for the dependencies of the driver. Whoever finds errors or omissions here can correct them in time.

ptpt52 commented:
we can find the hardware specifies and determine the dependencies on these sites:
https://wikidevi.com/wiki/
https://openwrt.org/toh/hwdata/

The following devices use mt76x0e:
dir-810l
tiny-ac
mzk-750dhp
tplink_c2-v1
tplink_c20-v1
ArcherC20i
ArcherMR200
tplink_c20-v4
tplink_tl-wr902ac-v3
rt-ac51u
gl-mt750
hc5761
hc5861 (not sure)

mkresin commented:
I've done the review for the mt7620 changes. Would you please split the mt7620 changes into the following commits:

1. remove mt76 for board which doen't have a mt76 wireless and pci disabled in the dts (like ex2700)
2. disable pci for board which don't have anything connected to pci (last topic in this comment)
3. select only the required mt76 driver(s)

Would you please fix the driver selection for the following board, which I added in the meantime:

- LR-25G001.dts

The following boards have a mt76 pci wireless but you PR doesn't add the driver to the image:

- MZK-EX750NP.dts

The following boards have pci enabled in the dts, do not have a minipci connector, don't have anything in the pci node that points to a rt2xx wireless and no mt76 driver selected.

Please double check these and disable/remove the pci nodes from the dts files:

- CS-QR10.dts
- DB-WRT01.dts
- MicroWRT.dts
- OY-0001.dts

ptpt52 committed:
mt76x8: select the matching mt76 driver and disable the default mt76 metapackage

 - select the matching mt76 driver for these devices:
    HiWiFi HC5861B
    Mercury MAC1200R v2.0
    Netgear AC1200 R6120
    Buffalo WCR-1166DS
    ZyXEL Keenetic Extra II

 Because every device has selected the corresponding mt76 driver
 we now disable the mt76 metapackage by default to make sure that
 other devices (those don't need mt76) avoid selecting unwanted packages

Signed-off-by: Chen Minqiang <ptpt52@gmail.com>

ptpt52 committed:
ramips: add missing pci wifi nodes to dts files

These devices has pci wifi but not defined in dts:
 HC5761
 HC5861
 TINY-AC

Signed-off-by: Chen Minqiang <ptpt52@gmail.com>

ptpt52 committed:
mt7620: select/deselect the matching mt76 driver and disable the default mt76 metapackage

 We can find the hardware specifies and determine the dependencies on these sites:
  https://wikidevi.com/wiki/
  https://openwrt.org/toh/hwdata/

 Because every device has selected the corresponding mt76 driver
 we now disable the mt76 metapackage by default to make sure that
 other devices (those don't need mt76) avoid selecting unwanted packages

Signed-off-by: Chen Minqiang <ptpt52@gmail.com>

ptpt52 commented:
In order to be clearer, can I split the commits like this?

ptpt52 commented:
> Please double check these and disable/remove the pci nodes from the dts files:
CS-QR10.dts
DB-WRT01.dts
MicroWRT.dts
OY-0001.dts

for these devices, they just enable pci for some (unknown/mistaken) reason, not about wifi, we can just ignore them for now.

mkresin commented:
> In order to be clearer, can I split the commits like this?

Close to what I have in mind but not exactly. I've done the split on my own and pushed the result to [my staging tree](https://git.openwrt.org/?p=openwrt/staging/mkresin.git;a=summary).

Would you please review the changes in [my staging tree](https://git.openwrt.org/?p=openwrt/staging/mkresin.git;a=summary) and leave me a note here if you are fine with what I've done.

I dropped the mt76x8 changes for now, as I haven't had the time to review these changes.

ptpt52 commented:
> Would you please review the changes in my staging tree and leave me a note here if you are fine with what I've done.
LGTM

mkresin commented:
Your "mt76x8: select the matching mt76 driver and disable the default mt76 metapackage" is wrong. mt7628 and mt7688 are wireless SoCs. Hence all boards required a wireless driver.

Unfortunately, my LinkIt 7688 is broken and I can't check which mt76 subdriver is required.

The good commits are in my staging tree and I'm going to close this as partially accepted.

ptpt52 commented:
@mkresin 

After further confirmation, yes, you are correct, the driver needed is mt7603e, so we can change the dependency to mt7603e and then remove mt76.
so you can directly modify it to your tree, or I should re-send a new PR?

ptpt52 commented:
it depends on
kmod-mt7603e
